pkg/api/user.go

```diff
@@ -29,8 +29,9 @@ type User struct {
 }
 
 // UserDetailer - used for mocking tests
+//go:generate mockery -name UserDetailer
 type UserDetailer interface {
-	GetUser(event *events.APIGatewayProxyRequest) *User
+	GetUser(reqCtx *events.APIGatewayProxyRequestContext) *User
```

Comment:
We don't have direct access to the `APIGatewayProxyRequest` object in our controllers, using the new gorilla mux adapter.